test/integration/odometry_publisher.cc

```diff
@@ -380,6 +380,86 @@ class OdometryPublisherTest
     EXPECT_NEAR(poses.back().Rot().Z(), tfPoses.back().Rot().Z(), 1e-2);
   }
 
+  /// \param[in] _sdfFile SDF file to load.
+  /// \param[in] _odomTopic Odometry topic.
+  protected: void TestMovement3dAtSingularity(const std::string &_sdfFile,
+                                              const std::string &_odomTopic)
+  {
+    // Start server
+    ServerConfig serverConfig;
+    serverConfig.SetSdfFile(_sdfFile);
+
+    Server server(serverConfig);
+    EXPECT_FALSE(server.Running());
+    EXPECT_FALSE(*server.Running(0));
+
+    // Create a system that records the body poses
+    test::Relay testSystem;
+
+    std::vector<math::Pose3d> poses;
+    testSystem.OnPostUpdate([&poses](const sim::UpdateInfo &,
+      const sim::EntityComponentManager &_ecm)
+      {
+        auto id = _ecm.EntityByComponents(
+          components::Model(),
+          components::Name("test_body"));
+        EXPECT_NE(kNullEntity, id);
+
+        auto poseComp = _ecm.Component<components::Pose>(id);
+        ASSERT_NE(nullptr, poseComp);
+        poses.push_back(poseComp->Data());
+      });
+    server.AddSystem(testSystem.systemPtr);
+
+    std::vector<math::Vector3d> odomAngVels;
+    // Create function to store data from odometry messages
+    std::function<void(const msgs::Odometry &)> odomCb =
+      [&](const msgs::Odometry &_msg)
+      {
+        odomAngVels.push_back(msgs::Convert(_msg.twist().angular()));
+      };
+
+    // Create node for publishing twist messages
+    transport::Node node;
+    auto cmdVel = node.Advertise<msgs::Twist>("/model/test_body/cmd_vel");
+    node.Subscribe(_odomTopic, odomCb);
+
+    // Set an angular velocity command that would cause pitch to update from 0
+    // to PI in 1 second, crossing the singularity when pitch is PI/2.
+    const math::Vector3d angVelCmd(0.0, M_PI, 0);
```

Comment:
```suggestion
    const math::Vector3d angVelCmd(0.0, GZ_PI, 0);
```

Reply:
Fixed in https://github.com/gazebosim/gz-sim/pull/2348/commits/90476c04124ee52ad07a602e6fe4df4547ad1a39

Reply:
Thanks!